lambdaclass/curse_of_mirra
# Player name - Create/Update

⚠️ Waiting new backend implementation 
Closes #1303 
Related Backend PR https://github.com/lambdaclass/game_backend/pull/138

## Motivation

Re implement name creation and update|

## Summary of changes
Players now can choose their username when the user is created in titlescreen. And update it later in mainscreen touching the display username section.

Created a new map field in SocketConnection to store all the players usernames.
Created a method to store the username in the DB ( `ServerUtils.SetUsername`)
Updated the `ServerUtils.CreateUser `method to also store the chosen username in the DB
Created PlayerNameHandler.UpdateUsername method To update the username
Created a string field username in LobbyConnection to store your client username.

## How has this been tested?

- Run the Backend. A popup field asking for a name should appear in the titlescreen. (if not your playerprefs has some values storaged, clean it up with PlayerPrefs.SetString("playerName",""))
- After you entered a name and pressed Enter, you should see in the backend console an insert query to the db with the username entered.
- Once you are in MainScreen you should see your username showing up in the top left corner. Press it to change it.
- Start a game and you should see your username displayed and also the BOTs. Also check the killfeed.


## Test additions / changes

List tests added/updated.

## Checklist
